### PR TITLE
Create types for entry

### LINKED ISSUE #1929: Entry: Define type interface for entry
### New Addition
* Define type interface for entry
* Add mock data for entries

### PR DESCRIPTION
**Pull request by samshara**
- fixes #1929

## Changes
* Add type for entry

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/newViews/Tagging/types.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    GeoLocationValue,
+    Matrix1dValue,
+    Matrix2dValue,
+    Types,
+} from '../AnalyticalFramework/types';
+
+
+interface BaseWidgetValue {
+    id: string;
+    title: string;
```
Comment (tnagorra) [inline src/newViews/Tagging/types.ts]: I don't know if we have these 2

File: src/newViews/Tagging/types.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    type: Types;
+}
+
+interface BaseValue<T> {
+    value: T;
+}
```
Comment (tnagorra) [inline src/newViews/Tagging/types.ts:25]: Isn't this too much of a nesting?

File: src/newViews/Tagging/types.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+export type Entry = {
+    id: number;
+    createdAt: string;
+    createdBy: number;
+    createdByName: string;
+    modifiedBy: number;
+    modifiedByName: string;
+    clientId: string;
+    versionId: number;
```
Comment (tnagorra) [inline src/newViews/Tagging/types.ts:116]: let's move these to a Entity type

File: src/newViews/Tagging/types.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    project: number;
+    lead: number;
+    analyticalFramework: number;
+    entryType: EntryType;
+    primaryTagging: SectionValue[];
+    secondaryTagging: WidgetValue[];
```
Comment (tnagorra) [inline src/newViews/Tagging/types.ts]: We should talk with navin about saving these.
We may also go the easier route and have widgets: WidgetData[] and then map them accordingly

File: src/newViews/Tagging/types.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    | ScaleWidgetValue
+    | OrganigramWidgetValue
+    | GeoLocationWidgetValue
+    | Matrix1dWidgetValue
+    | Matrix2dWidgetValue
+
+export interface SectionValue {
+    id: number;
+    title: string;
+    tooltip?: string;
```
Comment (tnagorra) [inline src/newViews/Tagging/types.ts]: IDK if these are necessary on SectionValue

File: src/newViews/Tagging/types.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    entryType: EntryType;
+    attributes: WidgetValue[];
+} | {
+    entryType: 'excerpt';
+    excerpt: string;
+}| {
+    entryType: 'image';
+    image: string;
+    imageRaw: string;
+}
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/types.ts]: Add for tabular as well

File: src/newViews/Tagging/types.ts [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    lead: number;
+    analyticalFramework: number;
+    entryType: EntryType;
+    attributes: WidgetValue[];
+} | {
+    entryType: 'excerpt';
+    excerpt: string;
+}| {
+    entryType: 'image';
+    image: string;
```
Comment (AdityaKhatri) [inline src/newViews/Tagging/types.ts:151]: We need imageDetails that contains a file and an id

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/newViews/Tagging/mockData.ts [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+                    },
+                },
+            },
+            {
+                id: 'widget9',
+                type: 'time-range',
+                data: {
+                    value: {
+                        startTime: '1627636952267',
+                        endTime: '1627637267565',
```
Comment (tnagorra) [inline src/newViews/Tagging/mockData.ts]: I think these should be number. Let's check with server as well.

File: src/newViews/Tagging/mockData.ts [hunk 9]
```diff
@@ -0,0 +1,154 @@
+import { Entry } from './types';
```
Comment (tnagorra) [inline src/newViews/Tagging/mockData.ts:1]: We should verify that the the entry information is conformant with analysis framework.
We should either wait for the AF mock data or fix this after it is merged

File: src/newViews/Tagging/mockData.ts (hunk 8 above)
Comment (samshara) [reply src/newViews/Tagging/mockData.ts]: @tnagorra I have written them as a string to conform with AF typings

Comment (samshara) [reply src/newViews/Tagging/mockData.ts]: fixed

**Review by tnagorra — APPROVED** [review]

**Pull request merged**